Trouble compiling on Mac OS X (10.10)

Salut,

I'm trying to get limbo up and running on a mac.  I've found the tutorials very well written, and I've successfully installed eigen, and boost (1.54) is pre-installed.  However, a waf build on my system hangs with the following error -- I'm including the output from configure as well (apologies for some of the weird formatting errors due to cut-and-paste from the X11 terminal into this text box)l:

`WARNING simplejson not found some function may not work
Setting top to                           : /Users/rieffelj/src/limbo-master 
Setting out to                           : /Users/rieffelj/src/limbo-master/build 
Checking for 'clang++' (C++ compiler)    : /usr/bin/clang++ 
Checking for 'clang' (C compiler)        : /usr/bin/clang 
Checking boost includes                  : 1_54 
Checking boost libs                      : ok 
Checking for Eigen                       : ok 
Checking Intel TBB includes              : not found 
Checking for compiler option to support OpenMP : Not supported 
Checking Intel MKL includes                    : Not found 
Checking for NLOpt includes                    : Not found 
Checking for libcmaes includes                 : Not found 
['-Wall', '-std=c++11', '-fdiagnostics-color', '-O3', '-march=native', '-g']
'configure' finished successfully (0.347s)
bash-3.2$ ./waf build
WARNING simplejson not found some function may not work
Waf: Entering directory `/Users/rieffelj/src/limbo-master/build'
[ 4/30] Compiling src/tests/bo_functions.cpp
[13/30] Compiling src/ehvi/ehvi_hvol.cc
[14/30] Compiling src/ehvi/ehvi_multi.cc
[15/30] Compiling src/ehvi/helper.cc
[16/30] Compiling src/hv/hypervol.c
[17/30] Linking build/src/liblimbo.a
[18/30] Linking build/src/tests/test_gp
[24/30] Linking build/src/tests/test_macros
[24/30] Linking build/src/tests/test_init_functions
[24/30] Linking build/src/examples/obs_multi_auto_mean
[24/30] Linking build/src/examples/obs_multi
[24/30] Linking build/src/tests/boptimizer
[24/30] Linking build/src/examples/mono_dim
[25/30] Linking build/src/tests/test_optimizers
[26/30] Processing build/src/tests/test_gp
[27/30] Processing build/src/tests/test_macros
[28/30] Processing build/src/tests/test_init_functions
[29/30] Processing build/src/tests/test_optimizers
In file included from ../src/tests/bo_functions.cpp:20:
/Users/rieffelj/src/limbo-master/src/limbo/bayes_opt/boptimizer.hpp:52:2: warning: NO NLOpt, and NO Libcmaes: the acquisition function will be optimized by a grid search algorithm (which is usually bad). Please install at least NLOpt or libcmaes to use limbo!. [-W#warnings]
#warning NO NLOpt, and NO Libcmaes: the acquisition function will be optimized by a grid search algorithm (which is usually bad). Please install at least NLOpt or libcmaes to use limbo!.
 ^
../src/tests/bo_functions.cpp:285:18: error: no template named 'map' in namespace 'std'; did you mean 'max'?
    typedef std::map<std::string,
            ~~~~~^~~
                 max
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/algorithm:2654:1: note: 'max' declared here
max(const _Tp& __a, const _Tp& __b, _Compare __comp)
^
../src/tests/bo_functions.cpp:285:18: error: definition or redeclaration of 'max' not allowed inside a function
    typedef std::map<std::string,
            ~~~~~^
../src/tests/bo_functions.cpp:285:18: error: C++ requires a type specifier for all declarations
    typedef std::map<std::string,
    ~~~~~~~      ^
../src/tests/bo_functions.cpp:285:18: error: typedef declarator cannot be qualified
    typedef std::map<std::string,
            ~~~~~^
../src/tests/bo_functions.cpp:285:18: error: typedef name must be an identifier
    typedef std::map<std::string,
                 ^~~~~~~~~~~~~~~~
../src/tests/bo_functions.cpp:285:21: error: expected ';' at end of declaration
    typedef std::map<std::string,
                    ^
                    ;
../src/tests/bo_functions.cpp:289:5: error: unknown type name 'res_t'
    res_t results;
    ^
1 warning and 7 errors generated.

Waf: Leaving directory `/Users/rieffelj/src/limbo-master/build'
Build failed
 -> task in 'bo_functions' failed (exit status 1): 
        {task 47707024: cxx bo_functions.cpp -> bo_functions.cpp.1.o}
['/usr/bin/clang++', '-Wall', '-std=c++11', '-fdiagnostics-color', '-O3', '-march=native', '-g', '-I/Users/rieffelj/src/limbo-master/build/src/tests', '-I/Users/rieffelj/src/limbo-master/src/tests', '-I/Users/rieffelj/src/limbo-master/build/src', '-I/Users/rieffelj/src/limbo-master/src', '-I/Users/rieffelj/src/limbo-master/build', '-I/Users/rieffelj/src/limbo-master', '-I/opt/local/include', '-I/usr/include/eigen3', '-I/usr/local/include/eigen3', '-I/usr/include', '-I/usr/local/include', '../src/tests/bo_functions.cpp', '-c', '-o', '/Users/rieffelj/src/limbo-master/build/src/tests/bo_functions.cpp.1.o']
`

A new issue:

`bash-3.2$ git status
On branch bo_fun_fix
Your branch is up-to-date with 'origin/bo_fun_fix'.
nothing to commit, working directory clean
bash-3.2$ ./waf build
WARNING simplejson not found some function may not work
Waf: Entering directory `/Users/rieffelj/src/limbo/limbo/limbo/build'
[ 4/30] Compiling src/tests/bo_functions.cpp
[13/30] Compiling src/ehvi/ehvi_hvol.cc
[14/30] Compiling src/ehvi/ehvi_multi.cc
[15/30] Compiling src/ehvi/helper.cc
[16/30] Compiling src/hv/hypervol.c
[17/30] Linking build/src/liblimbo.a
[18/30] Linking build/src/tests/test_gp
[20/30] Linking build/src/tests/test_init_functions
[23/30] Linking build/src/tests/test_macros
[24/30] Linking build/src/examples/mono_dim
[24/30] Linking build/src/examples/obs_multi_auto_mean
[24/30] Linking build/src/tests/test_optimizers
[24/30] Linking build/src/examples/obs_multi
[25/30] Linking build/src/tests/boptimizer
[27/30] Processing build/src/tests/test_macros
[28/30] Processing build/src/tests/test_gp
[28/30] Processing build/src/tests/test_optimizers
[29/30] Processing build/src/tests/test_init_functions
In file included from ../src/tests/bo_functions.cpp:19:
/Users/rieffelj/src/limbo/limbo/limbo/src/limbo/bayes_opt/boptimizer.hpp:52:2: warning: NO NLOpt, and NO Libcmaes: the acquisition function will be optimized by a grid search algorithm (which is usually bad). Please install at least NLOpt or libcmaes to use limbo!. [-W#warnings]
#warning NO NLOpt, and NO Libcmaes: the acquisition function will be optimized by a grid search algorithm (which is usually bad). Please install at least NLOpt or libcmaes to use limbo!.
 ^
../src/tests/bo_functions.cpp:270:18: error: no type named 'accessor' in 'std::__1::map<std::__1::basic_string<char>, std::__1::vector<std::__1::pair<double, double>, std::__1::allocator<std::__1::pair<double, double> > >, std::__1::less<std::__1::basic_string<char> >, std::__1::allocator<std::__1::pair<const std::__1::basic_string<char>, std::__1::vector<std::__1::pair<double, double>, std::__1::allocator<std::__1::pair<double, double> > > > > >'
    typename T1::accessor a;
    ~~~~~~~~~~~~~^~~~~~~~
../src/tests/bo_functions.cpp:299:17: note: in instantiation of function template specialization 'add_to_results<std::__1::map<std::__1::basic_string<char>, std::__1::vector<std::__1::pair<double, double>, std::__1::allocator<std::__1::pair<double, double> > >, std::__1::less<std::__1::basic_string<char> >, std::__1::allocator<std::__1::pair<const std::__1::basic_string<char>, std::__1::vector<std::__1::pair<double, double>, std::__1::allocator<std::__1::pair<double, double> > > > > >, std::__1::pair<double, double> >' requested here
                add_to_results("Sphere", results, std::make_pair(x_opt, opt.best_observation()(0)));
                ^
1 warning and 1 error generated.

Waf: Leaving directory `/Users/rieffelj/src/limbo/limbo/limbo/build'
Build failed
 -> task in 'bo_functions' failed (exit status 1): 
        {task 29873264: cxx bo_functions.cpp -> bo_functions.cpp.1.o}
['/usr/bin/clang++', '-Wall', '-std=c++11', '-fdiagnostics-color', '-O3', '-march=native', '-g', '-I/Users/rieffelj/src/limbo/limbo/limbo/build/src/tests', '-I/Users/rieffelj/src/limbo/limbo/limbo/src/tests', '-I/Users/rieffelj/src/limbo/limbo/limbo/build/src', '-I/Users/rieffelj/src/limbo/limbo/limbo/src', '-I/Users/rieffelj/src/limbo/limbo/limbo/build', '-I/Users/rieffelj/src/limbo/limbo/limbo', '-I/opt/local/include', '-I/usr/include/eigen3', '-I/usr/local/include/eigen3', '-I/usr/include', '-I/usr/local/include', '../src/tests/bo_functions.cpp', '-c', '-o', '/Users/rieffelj/src/limbo/limbo/limbo/build/src/tests/bo_functions.cpp.1.o']
`

--- a/src/tests/bo_functions.cpp
+++ b/src/tests/bo_functions.cpp
@@ -8,12 +8,11 @@
 #include <Eigen/Core>
 
 #ifdef USE_TBB
-#include <map>
-
 #include <tbb/task_scheduler_init.h>
 #include <tbb/parallel_for.h>
 #include <tbb/concurrent_hash_map.h>
 #else
+#include <map>
 #endif
 
 #include <limbo/tools/macros.hpp>

--- a/src/tests/bo_functions.cpp
+++ b/src/tests/bo_functions.cpp
@@ -267,9 +267,16 @@ bool is_in_argv(int argc, char** argv, const char* needle)
 template <typename T1, typename T2>
 void add_to_results(const char* key, T1& map, const T2& p)
 {
+#ifdef USE_TBB
     typename T1::accessor a;
     if (!map.find(a, key))
         map.insert(a, key);
+#else
+    typename T1::iterator a;
+    a = map.find(key);
+    if (a == map.end())
+        map[key] = std::vector<std::pair<double, double>>();
+#endif
     a->second.push_back(p);
 }
 

